lib/src/table_calendar_base.dart

```diff
@@ -153,7 +153,9 @@ class _TableCalendarBaseState extends State<TableCalendarBase>
       _pageCallbackDisabled = true;
     }
 
-    if (shouldAnimate && widget.pageAnimationEnabled) {
+    if (!_pageController.hasClients) {
+      _pageController = PageController(initialPage: currentIndex);
+    } else if (shouldAnimate && widget.pageAnimationEnabled) {
```

Comment:
Hey Michael, thanks for reporting the issue and working on it!

As for the fix, do you encounter any problems after simply wrapping this if...else... block with `if (_pageController.hasClients) { ... }`? This would allow the `_pageController` to stay `final` and, subsequently, prevent reinitialization of this field, which could lead to unexpected bugs in the future.

Reply:
Yeah, I had the same idea and tried that first, but it does not work: The wrong page (the one which with the `_pageController` was constructed) will be shown when the Element for the `TableCalendar` is finally mounted.

Reply:
I've done some testing and it seems to be the way to go about this issue. One more thing needed to be included - the updated `PageController` has to be supplied to the parent widget, otherwise the old instance will be incorrectly used. I'll probably have to add another callback for this, since using `onCalendarCreated()` to refresh `PageController` instance might be a bit misleading.

Can you verify if the current state of this PR solves issue #651?